Интерфейс задачи содержит следующий поля:
- тип(имя) запускаемой задачи
- флаг, разрешающий запускать задачу второй раз, не дожидаясь выполнения её в первый раз
- идентификатор задачи
- метод Run

Есть два варианта решения. 

1) С использованием библиотеки HangFire

Алгоритм:
- Пользователь выбирает тип запускаемой задачи
- Тип задачи передается в класс-менеджер
- Класс-менеджер запускает саму задачу через класс-executor
- Специальный фильтр, отслеживающий операции hangfire отлавливает событие запуска задачи и сохраняет информацию о ней в классе TaskMemory(класс отвечает за хранение информации о запущенной задаче)
- По завершению выполнения задачи фильтр отлавливает событие завершения задачи и удаляет информации о задаче из TaskMemory.
TaskMemory будет хранить информацию в виде Dictionary, ключом в которой будет идентификатор запущенной задачи.
Для того, чтобы отследить завершение задачи необходимо написать фильтр с двумя методами:
	-- OnPerforming(PerformingContext filterContext), запускаемый в начале выполнении задачи.  
	-- OnPerformed(PerformingContext filterContext), запускаемый в по завершению выполнения задачи.  
filterContext - информация о задаче(вернее о методе Run(), который находится внутри класса задачи).
Сложность заключается в том, как узнать информацию из filterContext о том, является ли задача одиночно выполняемой. При этом тип класса задачи можно узнать из MethodInfo, содержащийся в  filterContext. 
 
 2) Использование Task.Run
В этом случае у нас нет ui-функционала hangfire. Однако мы точно можем определить какая задача начинает свое выполнения и когда она завершится без использования фильтров.
Алгоритм:
- Пользователь выбирает тип запускаемой задачи
- Тип задачи передается в класс-менеджер
- Класс-менеджер запускает саму задачу через TaskRun и сохраняет информацию о запуске. При этом сохраняемые данные могут быть следующими:
	-- Идентификатор задачи
	-- Тип задачи
	-- Время начала
	-- Время окончаний
	-- Exception, в случае вы
- При завершении задачи следующие данные обновляются 
	-- проставляется время завершения задачи
	-- при возникновении исключений записывается сообщение об ошибке
Из сложностей - достаточно долгая реализация данного функционала.
